reporter_test.go

```diff
@@ -0,0 +1,29 @@
+package httpexpect
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockT struct {
+	testing.T
+}
+
+var fatalfInvoked = false
```

Comment:
Let's make this to be field of mockT, so that different instances of mockT wont affect each other.